src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -246,7 +248,7 @@ private async Task ForceAuthenticationAsync<TIOAdapter>(TIOAdapter adapter, bool
 
                 if (!receiveFirst)
                 {
-                    message = _context.NextMessage(reAuthenticationData, 0, (reAuthenticationData == null ? 0 : reAuthenticationData.Length));
+                    message = _context.NextMessage(reAuthenticationData == null ? default : new ReadOnlySpan<byte>(reAuthenticationData));
```

Comment:
This conditional shouldn't be necessary, as a null array is converted to a default span by span's ctor/implicit cast/etc.
```suggestion
                    message = _context.NextMessage(reAuthenticationData);
```